pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -1150,6 +1156,22 @@ func (dm *DockerManager) fmtDockerOpts(opts []dockerOpt) ([]string, error) {
 		sep = optSeparatorOld
 	}
 
+	return sep, nil
+}
+
+// fmtDockerOpts formats the given dockerOpts with the correct separator for
+// the docker API version or returns an error.
+func (dm *DockerManager) fmtDockerOpts(opts []dockerOpt) ([]string, error) {
```

Comment:
make this also take the separator so we don't parse twice? (call to fmtDockerOpts and ModifyHostConfig are in the same method)